Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -190,27 +190,18 @@ extension AWSClient {
 
     func nioRequestWithSignedHeader(_ nioRequest: Request) throws -> Request {
         var nioRequest = nioRequest
+
+        guard let url = URL(string: nioRequest.head.uri), let _ = url.hostWithPort else {
+            fatalError("nioRequest.head.uri is invalid.")
```

Comment:
I think we can/should use [our already-defined](https://github.com/swift-aws/aws-sdk-swift-core/pull/62/files#diff-7863205517fdde3415e746c5fee58bc6R27) `RequestError.invalidURL` if possible?